packages/yarnpkg-fslib/sources/xfs.ts

```diff
@@ -44,19 +43,24 @@ function registerCleanExit() {
   });
 }
 
+let tmpdir: PortablePath | null = null;
```

Comment:
Change this to const, as the value in `tmpdir` is not dependent on xfs object?
```suggestion
const tmpdir = npath.toPortablePath(os.tmpdir());
```

The lines 56 and 96 can be deleted.